projects/gfbio-app/src/app/oidc/oidc.component.ts

```diff
@@ -0,0 +1,68 @@
+import {Component, OnInit, Inject, OnDestroy} from '@angular/core';
+import {UserService, Config, User} from "wave-core";
+import {AppConfig} from "../app-config.service";
+import {first, SubscriptionLike} from "rxjs";
+
+@Component({
+    selector: 'wave-gfbio-login',
+    templateUrl: 'oidc.component.html',
+    styleUrls: ['./oidc.component.scss'],
+})
+export class OidcComponent implements OnInit, OnDestroy {
+    private userSubscription : SubscriptionLike | undefined;
+    private loginSubscription : SubscriptionLike | undefined;
+    private logoutSubscription : SubscriptionLike | undefined;
+
+    user? : User;
+
+    constructor(
+        @Inject(Config) private readonly config: AppConfig,
+        private readonly userService: UserService,) {
+    }
+
+    ngOnInit(): void {
+        this.userSubscription = this.userService
+            .getSessionStream()
+            .pipe(first())
+            .subscribe((session) => {
+                if (!session.user || session.user.isGuest) {
+                } else {
+                    this.user = session.user;
+                }
+            });
+    }
+
+    login() : void {
+        this.loginSubscription = this.userService.oidcInit().subscribe((idr) => {
```

Comment:
and maybe here

Reply:
Especially if you allow calling this function and thus creating it several times, you would either need to destroy it at the header of the function… or you don't need to unsubscribe after all by using something like `first`, etc.

Reply:
I disabled the button while a request is pending. A BehaviorSubject seemed like the best way, but if there is a better one, let me know. c136f37f0518115ce6a71ddb29722e6b0cb70703 28a3bec946cc25c3de364287db08fa2edd3646c5